subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1394,6 +1395,26 @@ static int __wifi_ap_config_args_to_params(const struct shell *sh, size_t argc,
 			params->max_inactivity = (uint32_t)val;
 			params->type |= WIFI_AP_CONFIG_PARAM_MAX_INACTIVITY;
 			break;
+		case 's':
+			val = shell_strtol(optarg, 10, &err);
+			if (err) {
+				shell_error(sh, "Unable to parse input for max_num_sta (err %d)",
+					    err);
+				return err;
+			}
+			if (val < 0) {
+				PR_ERROR("Invalid value for max_num_sta %ld", val);
+				return -EINVAL;
+			}
```

Comment:
We are duplicating this check multiple times, could we refactor this to a checker function that could be called from different case statements.